src/mc-range-encoding.c

```diff
@@ -345,42 +442,146 @@ static mlib_int128 dec128_to_int128(mc_dec128 dec) {
     return ret;
 }
 
-bool mc_canUsePrecisionModeDecimal(mc_dec128 min, mc_dec128 max, uint32_t precision, uint32_t *maxBitsOut) {
+#define SIGNED_INT_128_MAX_DECIMAL mc_dec128_from_string("170141183460469231731687303715884105727")
+#define UNSIGNED_INT_128_MAX_DECIMAL mc_dec128_from_string("340282366920938463463374607431768211455")
```

Comment:
Add comments for context. Also (nit), I prefer we use "(U)INT128_MAX_AS_DECIMAL128" for the naming.
```suggestion
// (2^127 - 1) = the maximum signed 128-bit integer value, as a decimal128
#define INT128_MAX_AS_DECIMAL128 mc_dec128_from_string("170141183460469231731687303715884105727")
// (2^128 - 1) = the max unsigned 128-bit integer value, as a decimal128
#define UINT128_MAX_AS_DECIMAL128 mc_dec128_from_string("340282366920938463463374607431768211455")
```

Reply:
Done.